t/43-scheduling-and-worker-scalability.t

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env perl
+# Copyright (C) 2019-2020 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+use strict;
+use warnings;
+
+use FindBin;
+use lib "$FindBin::Bin/lib";
+use OpenQA::Scheduler::Model::Jobs;
+use OpenQA::Test::Database;
+use OpenQA::Jobs::Constants;
+use OpenQA::Test::Utils
+  qw(create_user_for_workers create_webapi setup_share_dir create_websocket_server stop_service setup_fullstack_temp_dir);
+use Test::More;
+use Test::Warnings;
+use Time::HiRes 'sleep';
+use File::Path 'make_path';
+use Scalar::Util 'looks_like_number';
+use Mojo::File 'path';
+
+BEGIN {
+    $ENV{OPENQA_SCHEDULER_MAX_JOB_ALLOCATION} = $ENV{SCALABILITY_FULLSTACK_JOB_COUNT}
```

Comment:
That environment variable is used to initialize a constant. Therefore it needs to be set within `BEGIN` to have effect. (Correct me if I'm wrong.)

It is set at all so the scheduler will assign all the jobs at once and there's no need to run multiple scheduler ticks. That simplifies the assertions within the test. But it is still a draft PR because I still need to think about that details at this point, too.